src/haywire/http_server.c

```diff
@@ -378,6 +385,132 @@ void http_stream_on_read_http_parser(uv_stream_t* tcp, ssize_t nread, const uv_b
     }
 }
 
+void http_stream_on_read_pico(uv_stream_t* tcp, ssize_t nread, const uv_buf_t* buf)
+{
+    http_connection* connection = (http_connection*)tcp->data;
+
+    if (nread > 0)
+    {
+        // TODO: Should this be at a higher level? Both parsers call this the same way.
+        http_request_buffer_consume(connection->buffer, nread);
+
+        printf("START\tNREAD: %d BUFUSED: %d PREVBUFLEN: %d\n",
+               nread,
+               http_request_buffer_get_used(connection->buffer),
+               connection->prevbuflen);
+        //fwrite(http_request_buffer_get_buffer(connection->buffer),
+        //       http_request_buffer_get_used(connection->buffer), 1, stdout);
+
+        while (connection->prevbuflen < http_request_buffer_get_used(connection->buffer))
```

Comment:
@kellabyte I don't get this while loop. Is this so you loop through multiple requests potentially read together? shouldn't it be `while(total_parsed < nread)` , where `total_parsed` is incremented after the call to `phr_parse_request`?